lib/msf/core/modules/metadata/search.rb

```diff
@@ -34,7 +34,7 @@ def self.parse_search_string(search_string)
     search_string ||= ''
     search_string += ' '
     
-    search_string = search_string.gsub(/((\s*?\:+)+$)|((\:*\s)+$)/,'')
+    search_string = search_string.gsub(/((\:+\s*)\z|(\\s*:+)\z|(\:\s*:+ *)\z)/,'')
```

Comment:
My only comment on this is that you seem to be using one more capture group that is necessary for this regex to work, particularly given that we aren't interested in using any of the capture groups later on in the code. 

Also you seem to have an extra backslash here, so it should be `\s*:+`. Right now the regex for the middle expression is set up to explicitly match on `\s` as a string, reoccuring 0 or more times, followed by an `:` one or more times.

Proposed change below:
```suggestion
    search_string = search_string.gsub(/(\:+\s*)\z|(\s*:+)\z|(\:\s*:+ *)\z/,'')
```

Reply:
Upon further review I'm also not sure what the purpose of the last capture group is? It doesn't seem to fit in with what was originally the issue aka the trailing `:` character. Mind explaining what the purpose of that was?

Reply:
Alright yeah so even with that update to the regex you did, your third capture group will still fail to capture things like `use exploit/linux/local/nested_namespace_idmap_limit_priv_e:                                      :                 :`. Seems the regex needs a bit of updating.

Reply:
Ok found a solution, seems like `(\:+(\s*\:*)+\z)|(\s+(\s*\:*)+\z)` should resolve this. Here are some tests for reference:

```
msf6 > use exploit/linux/local/nested_namespace_idmap_limit_priv_e:::: :
"exploit/linux/local/nested_namespace_idmap_limit_priv_e:::: :  "
[*] Using configured payload linux/x86/meterpreter/reverse_tcp

Matching Modules
================

   #  Name                                                       Disclosure Date  Rank   Check  Description
   -  ----                                                       ---------------  ----   -----  -----------
   0  exploit/linux/local/nested_namespace_idmap_limit_priv_esc  2018-11-15       great  Yes    Linux Nested User Namespace idmap Limit Local Privilege Escalation


Interact with a module by name or index. For example info 0, use 0 or use exploit/linux/local/nested_namespace_idmap_limit_priv_esc

[*] Using exploit/linux/local/nested_namespace_idmap_limit_priv_esc
msf6 exploit(linux/local/nested_namespace_idmap_limit_priv_esc) > use exploit/linux/local/nested_namespace_idmap_limit_priv_e                                                   :
"exploit/linux/local/nested_namespace_idmap_limit_priv_e :  "
[*] Using configured payload linux/x86/meterpreter/reverse_tcp

Matching Modules
================

   #  Name                                                       Disclosure Date  Rank   Check  Description
   -  ----                                                       ---------------  ----   -----  -----------
   0  exploit/linux/local/nested_namespace_idmap_limit_priv_esc  2018-11-15       great  Yes    Linux Nested User Namespace idmap Limit Local Privilege Escalation


Interact with a module by name or index. For example info 0, use 0 or use exploit/linux/local/nested_namespace_idmap_limit_priv_esc

[*] Using exploit/linux/local/nested_namespace_idmap_limit_priv_esc
msf6 exploit(linux/local/nested_namespace_idmap_limit_priv_esc) > use exploit/linux/local/nested_namespace_idmap_limit_priv_e  :                  :     ::::                 :
"exploit/linux/local/nested_namespace_idmap_limit_priv_e : : :::: :  "
[*] Using configured payload linux/x86/meterpreter/reverse_tcp

Matching Modules
================

   #  Name                                                       Disclosure Date  Rank   Check  Description
   -  ----                                                       ---------------  ----   -----  -----------
   0  exploit/linux/local/nested_namespace_idmap_limit_priv_esc  2018-11-15       great  Yes    Linux Nested User Namespace idmap Limit Local Privilege Escalation


Interact with a module by name or index. For example info 0, use 0 or use exploit/linux/local/nested_namespace_idmap_limit_priv_esc

[*] Using exploit/linux/local/nested_namespace_idmap_limit_priv_esc
msf6 exploit(linux/local/nested_namespace_idmap_limit_priv_esc) > 
```